code/datums/components/light_eater.dm

```diff
@@ -0,0 +1,198 @@
+/**
+ * Makes anything it attaches to capable of removing something's ability to produce light until it is destroyed
+ *
+ * The permanent version of this is [/datum/element/light_eater]
+ */
+/datum/component/light_eater
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	/// Tracks things this light eater has eaten
+	var/list/eaten_lights
+
+/datum/component/light_eater/Initialize(list/_eaten)
+	if(!isatom(parent) && !istype(parent, /datum/reagent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(length(_eaten))
+		eaten_lights = list()
+		var/list/cached_eaten_lights = eaten_lights
+		for(var/food in _eaten)
+			RegisterSignal(food, COMSIG_ATOM_UPDATE_LIGHT, .proc/block_light_update)
+			RegisterSignal(food, COMSIG_PARENT_QDELETING, .proc/deref_eaten_light)
+			RegisterSignal(food, COMSIG_PARENT_EXAMINE, .proc/on_eaten_light_examine)
+			cached_eaten_lights += food
+	return ..()
```

Comment:
```suggestion
/datum/component/light_eater/Initialize(list/_eaten)
	. = ..()
	if(!isatom(parent) && !istype(parent, /datum/reagent))
		return COMPONENT_INCOMPATIBLE

	if(length(_eaten))
		eaten_lights = list()
		var/list/cached_eaten_lights = eaten_lights
		for(var/food in _eaten)
			RegisterSignal(food, COMSIG_ATOM_UPDATE_LIGHT, .proc/block_light_update)
			RegisterSignal(food, COMSIG_PARENT_QDELETING, .proc/deref_eaten_light)
			RegisterSignal(food, COMSIG_PARENT_EXAMINE, .proc/on_eaten_light_examine)
			cached_eaten_lights += food

```
Call parent first on initialization whenever you can.

Reply:
Done